python/tank/pipelineconfig_factory.py

```diff
@@ -279,29 +278,38 @@ def _get_configuration_context():
         # Once found, we can double check that the current Entity is actually
         # associated with the project that the PC is associated with.
         val = pipelineconfig_utils.get_config_install_location(curr_pc_path)
-        
+
     return val
 
 
+def _cleanup_separators(path):
+    """
+    Flips all slashes to the operating system's standard slash separator and strips
```

Comment:
@benoit-leveau Thanks for the suggestion Benoît.

I was actually more thinking of something that is outside of the python/{tank,sgtk} folder and more into python/shared or python/tank_vendor. Moving forward, we want to take bits and pieces of the core (app store code for example) and make them standalone modules to make more and more of Toolkit's code reusable outside Toolkit or configured pipeline. Those bits and pieces would be broken up into independent submodules that would reside inside tank_vendor or some other location.

A good example of this approach is the tank_vendor.shotgun_authentication module that we introduced in 0.16. While we are only packaging it as part of the core, it is actually a fully independent module that anybody who uses the Shotgun API but not Toolkit can also use. The Shotgun Desktop actually uses that module in order to bootstrap Toolkit.